stablehlo/conversions/linalg/transforms/StablehloToLinalgDotProduct.cpp

```diff
@@ -186,11 +186,6 @@ struct DotGeneralOpConversion final
       if (succeeded(lowerDotOp<DotGeneralOp, linalg::DotOp>(
               rewriter, getTypeConverter(), op, adaptor)))
         return success();
-      std::string str;
-      llvm::raw_string_ostream os(str);
-      os << "supposedly simple DotGeneralOp could not be converted: ";
```

Comment:
This is all in a branch for `if (op.isSimpleDot())`, should that method be updated instead? Or are there cases where a simple dot lowering should fall back to dot general lowering?

Reply:
Maybe update `isSimpleDot()` is more suitable. Either one is fine for me.

Reply:
Let's update `isSimpleDot`. What would the change be? Adding a rank check?

We have [legalize_dot_general_to_dot.cc](https://github.com/openxla/xla/blob/4fbe1339c2957aaa7a18a5b594d7ee2257567260/xla/mlir_hlo/mhlo/transforms/legalize_dot_general_to_dot/legalize_dot_general_to_dot.cc#L53) which seems to require this.